plugins/src/ios/OptimoveSDKPlugin.swift

```diff
@@ -34,4 +34,105 @@ import NotificationCenter
+            return
+        }
+        
+        let screenCategory = command.arguments[1] as? String
+       
+        Optimove.shared.reportScreenVisit(screenTitle: screenTitle, screenCategory: screenCategory)
+        
+        self.commandDelegate.run {
+            let pluginResult = CDVPluginResult(status: .ok)
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+        }
+    }
+    
+    @objc(registerUser:)
+    func registerUser(command: CDVInvokedUrlCommand) {
+        guard let sdkId = command.arguments.first as? String else {
+            let pluginResult = CDVPluginResult(status: .error, messageAs: "id is invalid")
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+            return
+        }
+        
+        guard let  email = command.arguments[1] as? String else {
+            let pluginResult = CDVPluginResult(status: .error, messageAs: "email is invalid")
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+            return
+        }
+        
+        Optimove.shared.registerUser(sdkId: sdkId, email: email)
+        
+        self.commandDelegate.run {
+            let pluginResult = CDVPluginResult(status: .ok)
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+        }
+    }
+    
+    @objc(setUserId:)
+    func setUserId(command: CDVInvokedUrlCommand) {
+        guard let id = command.arguments.first as? String else {
+            let pluginResult = CDVPluginResult(status: .error, messageAs: "id is invalid")
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+            return
+        }
+        
+        Optimove.shared.setUserId(id)
+        
+        self.commandDelegate.run {
+            let pluginResult = CDVPluginResult(status: .ok)
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+        }
+    }
+    
+    @objc(getVisitorID:)
+    func getVisitorID(command: CDVInvokedUrlCommand) {
+        self.commandDelegate.run {
+            let pluginResult = CDVPluginResult(status: .ok, messageAs: Optimove.getVisitorID())
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+        }
+    }
+    
+    @objc(setUserEmail:)
+    func setUserEmail(command: CDVInvokedUrlCommand) {
+        guard let email = command.arguments.first as? String else {
+            let pluginResult = CDVPluginResult(status: .error, messageAs: "email is invalid")
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+            return
+        }
+        
+        Optimove.shared.setUserEmail(email: email)
+        
+        self.commandDelegate.run {
+            let pluginResult = CDVPluginResult(status: .ok)
+            self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+        }
+    }
 }
```

Comment:
Are we missing an implementation for getCurrentUserIdentifier ?

Reply:
OK, can you check this with Tom.

If that's true, I'm not sure we should expose it in JS at all.

The plugin will (ideally) have a unified interface in JS for both platforms.

Reply:
sure

Reply:
will be handled in js

Reply:
removed from js file